openpype/hosts/blender/plugins/create/create_model.py

```diff
@@ -21,29 +21,33 @@ def process(self):
         ops.execute_in_main_thread(mti)
 
     def _process(self):
-        # Get Instance Container or create it if it does not exist
-        instances = bpy.data.collections.get(AVALON_INSTANCES)
-        if not instances:
-            instances = bpy.data.collections.new(name=AVALON_INSTANCES)
-            bpy.context.scene.collection.children.link(instances)
 
-        # Create instance object
+        # get info from data and create name value
         asset = self.data["asset"]
         subset = self.data["subset"]
-        name = plugin.asset_name(asset, subset)
-        asset_group = bpy.data.objects.new(name=name, object_data=None)
-        asset_group.empty_display_type = 'SINGLE_ARROW'
-        instances.objects.link(asset_group)
+        #name = plugin.asset_name(asset, subset)
+
+        name = MODEL_TASK_NAME
+        # Get Instance Container or create it if it does not exist
+        instance = bpy.data.collections.get(name)
+        if not instance:
+            instance = bpy.data.collections.new(name=name)
+            bpy.context.scene.collection.children.link(instance)
+
+        #add custum property on the instance with the data
         self.data['task'] = api.Session.get('AVALON_TASK')
-        lib.imprint(asset_group, self.data)
+        lib.imprint(instance, self.data)
 
         # Add selected objects to instance
         if (self.options or {}).get("useSelection"):
-            bpy.context.view_layer.objects.active = asset_group
             selected = lib.get_selection()
             for obj in selected:
-                obj.select_set(True)
-            selected.append(asset_group)
-            bpy.ops.object.parent_set(keep_transform=True)
-
-        return asset_group
+                instance.objects.link(obj)
+                bpy.context.scene.collection.objects.unlink(obj)
+        else:
```

Comment:
I don't get the difference between `if` and `else` here. The code seems to be DRYable.